pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -327,32 +326,51 @@ func (p *DefaultProvider) blockDeviceMappings(blockDeviceMappings []*v1.BlockDev
 	return blockDeviceMappingsRequest
 }
 
+func convertToNil(number *int64) *int32 {
```

Comment:
nit: Drop in favor of https://github.com/samber/lo?tab=readme-ov-file#emptyabletoptr